### ISSUE TITLE
ARM version returns `Unsupported Architecture aarch64` on Apple M1 Silicon

### ISSUE THREAD

**Issue by rofrano** [IBM-Cloud/ibm-cloud-cli-release#180]
@steveclay  I was excited to see your announcement of an ARM version but was soon disappointed that I still get the following  error: `Unsupported Linux architecture: aarch64. Quit installation.` Apparently the `install` script was never updated.

### Operating environment 

Apple M1 Silicon Mac mini base model 8cpu/8gb memory

### What did you try?

Running inside an Ubuntu 21.04 VM using `multipass`.

```bash
curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
```

### What did you see?

```bash
Unsupported Linux architecture: aarch64. Quit installation.
```

### What did you expect to see?

I expect the ARM version to be installed.

I also tried:

```bash
$ wget https://download.clis.cloud.ibm.com/ibm-cloud-cli/2.6.0/IBM_Cloud_CLI_2.6.0_arm64.tar.gz
$ tar xzvf IBM_Cloud_CLI_2.6.0_arm64.tar.gz 
$ cd Bluemix_CLI/
$ sudo ./install
Unsupported Architecture aarch64
```

So still unsupported!

### Steps to Reproduce

Using `Multipass` on Apple M1 Silicon (Mac mini base model). Install Multipass if you don't have it.

```bash
brew install multipass
```

Bring up an ARM VM and try and install IBM Cloud CLI

```bash
multipass launch 21.10 -n impish
multipass shell impish
```

Make sure you have an ARM VM:

```bash
$ uname -a
Linux impish 5.13.0-35-generic #40-Ubuntu SMP Mon Mar 7 08:06:10 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

Run the installer:

```bash
$ curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
Unsupported Linux architecture: aarch64. Quit installation.
```

When looking at the `install` script there is no case statement for ARM. You just need to insert `aarch64 `:

```
MARCH=$(uname -m | tr '[:upper:]' '[:lower:]')
case "$MARCH" in
"aarch64") ;;
"ppc64le") ;;
"x86_64") ;;
"x86_32") ;;
"s390x") ;;
*)
```

**Comment by steveclay** [comment]
Thank you for bringing this to our attention. Looking into the installer, but I also wanted to point out that the combination you are attempting is not actually supported by the CLI. On a Mac M1, you will need to use the Mac ARM version of the CLI.

**Comment by rofrano** [comment]
> but I also wanted to point out that the combination you are attempting is not actually supported by the CLI. 

Why would I care about running `ibmcloud` natively on my M1 Mac? The Rosetta 2 version already works perfectly fine.

I, like many cloud developers, spend all day developing in Linux containers and in Linux VMs. On an Intel Mac those were x64 containers and VMs but now on an Apple M1 Mac those are ARM containers and ARM VMs which is why it was so critically important to have the IBM Cloud CLI running natively on ARM Linux.

Anyone working in Visual Studio Code using Remote Containers or Remote SSH will need native ARM support for Linux on Apple M1 Silicon because all of the containers and VM's will be ARM based. I'm shocked that your not supporting this as I don't know any developers that work natively on their Mac (since Mac is BSD unix not Linux). Most work in virtual environments. IMHO, those should be support (and tested).

**Comment by schmidtp0740** [comment]
@rofrano We updated the installer script for Linux to allow `aarch64`/`arm64` on our staging repo. Could you test and verify that the installer works by running the following command while on the IBM Network/Cisco Anyconnect? If it works then we will promote the change to our production repo
```
curl -fsSL https://download.clis.test.cloud.ibm.com/ibm-cloud-cli-installer-scripts/linux | sh
```

**Comment by rofrano** [comment]
Hi @schmidtp0740 Unfortunately it does not because you did not update the installer inside the package as I had pointed out in my post above. :( 

My Apple M1 Mac is not on the IBM Network (it is my personal development machine) but I did test it with an ARM instance of my Docker provider for Vagrant on my Intel Mac which is on the IBM Network and it failed using an ARM docker container.

### Steps to reproduce

You can perform this test yourself. Bring up an ARM based Linux container. I'm using the [rofrano/vagrant-provider:ubuntu](https://github.com/rofrano/vagrant-docker-provider) image that I use for the labs in my DevOps course at NYU but you can use a base `ubuntu` container and get the same result.

```bash
docker run --rm -it -h nyu --platform=linux/arm64 rofrano/vagrant-provider:ubuntu bash
root@nyu:/# 
```

That instantiated a ARM base Linux container. Let's verify that it is really ARM base using `name -a`:

```bash
root@nyu:/# uname -a
Linux nyu 5.10.104-linuxkit #1 SMP Wed Mar 9 19:05:23 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

Yup, we get back `aarch64` so this should work. Now use your command:

```bash
root@nyu:/# curl -fsSL https://download.clis.test.cloud.ibm.com/ibm-cloud-cli-installer-scripts/linux | sh
Current platform is linux-arm64. Downloading corresponding IBM Cloud CLI...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9.9M  100  9.9M    0     0  5121k      0  0:00:01  0:00:01 --:--:-- 5149k
Download complete. Executing installer...
Bluemix_CLI/
Bluemix_CLI/bin/
Bluemix_CLI/bin/ibmcloud
Bluemix_CLI/bin/ibmcloud.sig
Bluemix_CLI/bin/NOTICE
Bluemix_CLI/bin/LICENSE
Bluemix_CLI/bin/CF_CLI_Notices.txt
Bluemix_CLI/bin/CF_CLI_SLC_Notices.txt
Bluemix_CLI/autocomplete/
Bluemix_CLI/autocomplete/bash_autocomplete
Bluemix_CLI/autocomplete/zsh_autocomplete
Bluemix_CLI/install
Bluemix_CLI/uninstall
Bluemix_CLI/install_bluemix_cli
Unsupported Architecture aarch64
Install failed.
```

### Failed!
Unfortunately it still is not recognizing `aarch64` as a valid architecture. It looked hopeful when it said: `Current platform is linux-arm64. Downloading corresponding IBM Cloud CLI...` so the download script correctly identified `aarch64` and downloaded the correct package, but the second installer in the `Bluemix_CLI` package was not updated! 

You need to update the installer inside the `Bluemix_CLI/` package. It still reads:

**Bluemix_CLI/install**:
```bash
MARCH=$(uname -m | tr '[:upper:]' '[:lower:]')
case "$MARCH" in
"ppc64le") ;;
"x86_64") ;;
"x86_32") ;;
"s390x") ;;
*)
```

Hopefully you can use this procedure to test it yourself.

BTW, This is exactly how I would be using the `ibmcloud` command. See my article:  [Developing on Apple M1 Silicon with Virtual Environments](https://johnrofrano.medium.com/developing-on-apple-m1-silicon-with-virtual-environments-4f5f0765fd2f) for how I use Vagrant and Docker to create ARM based containers that behave like VM's to develop in and run commands like `ibmcloud` which, until now, have not worked on Apple M1 Macs. This is how the students in my class access the IBM Cloud on Intel Macs and I'm really hoping this solves that problem of not working on Apple Silicon Macs because every semester more and more students are showing up with M1 Macs (approx 1/4 the class).

**Comment by steveclay** [comment]
This should be resolved in release 2.7.0.
